protos/file.proto

```diff
@@ -28,10 +212,29 @@ message Metadata {
 
   // Position of the manifest in the file. If it is zero, the manifest is stored
   // externally.
+  //
+  // In Lance version 1 this is a protobuf-encoded Manifest message from the table
+  // format.
+  //
+  // In Lance version 2 this is a FileDescriptor message
+  //
+  // This message is immediately followed by the metadata block.  Therefore, the
+  // size of this block can be calculated by subtracting this value from the 
+  // metadata offset (that was used to read this message)
+  //
+  // This message is optional.  If a file is not self-describing then this must
+  // point to the start of the metadata block (e.g. representing a manifest with
+  // size 0)
```

Comment:
What would we think of making the `FileDescriptor` not optional for Lance v2? IMO it would make the format easier to debug if individual files were readable independently.

Reply:
That's fine with me.  It shouldn't be too large and readers don't have to read it so I don't see any downside to making it mandatory.

Reply:
I've changed the language to reinforce that this is required in v2 files.